### PR TITLE
WT-8796 Clone the correct mongo branch based on the wiredtiger branch for many-collection-test

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -60,7 +60,14 @@ functions:
...
         set -o errexit
         set -o verbose
-        git clone https://github.com/mongodb/mongo
+        mongo_repo=https://github.com/mongodb/mongo
+        if [ "${branch_name}" = "mongodb-5.0" ]; then
+          git clone $mongo_repo -b v5.0
+        elif [ "${branch_name}" = "mongodb-4.4" ]; then
+          git clone $mongo_repo -b v4.4
+        else
+          git clone $mongo_repo
```
Comment (etienneptl) [inline test/evergreen.yml:69]: Should we precise the `master` branch?

**Review by etienneptl — APPROVED** [review]
LGTM, one comment

File: test/evergreen.yml (hunk 1 above)
Comment (Siddhartha8899) [reply test/evergreen.yml]: Updated in [a6bf607](https://github.com/wiredtiger/wiredtiger/pull/7603/commits/a6bf607d7d1052c66358e51f4853ecf64d2f92db)

File: test/evergreen.yml [hunk 2]
```diff
@@ -57,10 +57,21 @@ functions:
...
         set -o errexit
         set -o verbose
-        git clone https://github.com/mongodb/mongo
+        mongo_repo=https://github.com/mongodb/mongo
+        if [[ "${branch_name}" == "mongodb-5.0" ]]; then
+          git clone $mongo_repo -b v5.0
+        elif [[ "${branch_name}" == "mongodb-4.4" ]]; then
+          git clone $mongo_repo -b v4.4
+        elif [[ "${branch_name}" =~ "mongodb-" ]]; then   
+          echo "Checkout the correct mongodb branch for wiredtiger branch ${branch_name}"
```
Comment (ajmorton) [inline test/evergreen.yml:70]: Thanks for the change!
Maybe reword the `echo` text as `'fetch mongo repo' does not currently handle branch ${branch_name}` and we're good to go.

Comment (etienneptl) [reply test/evergreen.yml:70]: Yep, I agree

**Review by etienneptl — APPROVED** [review]
LGTM, pending @ajmorton's suggestion.

File: test/evergreen.yml (hunk 2 above)
Comment (Siddhartha8899) [reply test/evergreen.yml]: Updated in [1e74f3a](https://github.com/wiredtiger/wiredtiger/pull/7603/commits/1e74f3a22e65144438bc1e480679b3426233503d)
